### PR TITLE
show more button and limit videos scrolls

### PR DESCRIPTION
**Pull request by mwilkins74**
Limited videos that are displayed on a video lists page start at 3. 
Implemented a `ShowMoreButton` that then shows 3 more. 
The button continues to render until the User hits the end of the current `VideoList`. 

'KEEP SHOWING MORE' button:
Screen Shot 2024-03-08 at 3 41 48 PM

Added `READY TO HIRE ME?` button below the 'KEEP SHOWING MORE'. 
Links to Contact component.
Figma calls for 'KEEP SHOWING MORE' to be below the `READY TO HIRE ME?` button, but I think logically the reverse makes more sense from a UI perspective. 
Screen Shot 2024-03-08 at 4 01 01 PM

### CODE REVIEW

File: src/components/organisms/videoList/VideoList.tsx [hunk 1]
```diff
@@ -2,27 +2,56 @@ import { Grid } from '@mui/material'
...
+import { useState } from 'react'
 
 type VideoListProps = {
   videos: IVideoInfo[]
   pageType: 'genre' | 'role' | 'recent'
 }
 
 export function VideoList({ videos, pageType }: VideoListProps) {
-  const videoComponents = videos.map(video => (
+  const videosPerPage = 3
```
Comment (ChuckBTaylor) [inline src/components/organisms/videoList/VideoList.tsx:15]: 🔧 Define this outside of component

File: src/components/organisms/videoList/VideoList.tsx [hunk 2]
```diff
@@ -2,27 +2,56 @@ import { Grid } from '@mui/material'
...
+    const currentVisibleCount = visibleVideos.length
+    const nextVisibleVideos = videos.slice(
+      currentVisibleCount,
+      currentVisibleCount + videosPerPage
+    )
+    setVisibleVideos((prevVisibleVideos: IVideoInfo[]) => [
+      ...prevVisibleVideos,
+      ...nextVisibleVideos,
+    ])
+  }
```
Comment (ChuckBTaylor) [inline src/components/organisms/videoList/VideoList.tsx:30]: :art:  This looks like it'll work. But just looks like a lot. We can just overwrite the whole list.

```suggestion
  const handleShowMore = () => {
    setVisibleVideos(videos.slice(0, visibleVideos.length + videosPerPage))
  }
```

File: src/components/organisms/videoList/VideoList.tsx [hunk 3]
```diff
@@ -2,27 +2,56 @@ import { Grid } from '@mui/material'
...
+          padding: '32px 16px 16px',
+          fontSize: '1.75rem',
+          fontWeight: '400',
+        }}
       />
       <Grid container spacing={2}>
         {videoComponents}
       </Grid>
+      {!allVideosDisplayed && <ShowMoreButton onClick={handleShowMore} />}
+      <HireMeButton />
```
Comment (ChuckBTaylor) [inline src/components/organisms/videoList/VideoList.tsx:54]: 🔧 Figma has HireMe above ShowMore

File: src/components/atoms/buttons/HireMeButton.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import { Button } from '@mui/material'
+import { Link, useLocation } from 'react-router-dom'
+
+export function HireMeButton() {
+  const location = useLocation()
+  const isHomePage = location.pathname === '/'
+
+  if (isHomePage) {
+    return null
+  }
```
Comment (ChuckBTaylor) [inline src/components/atoms/buttons/HireMeButton.tsx:10]: 🔧  I don't like the idea of an atom making this decision. Whether a component is rendered or not should come from its parent.

**Review by ChuckBTaylor — CHANGES_REQUESTED** [review]
Really cool. Just a couple feedback items.

File: src/components/organisms/videoList/VideoList.tsx (hunk 3 above)
Comment (mwilkins74) [reply src/components/organisms/videoList/VideoList.tsx:54]: Ok, I guess we want to prioritize Hiring over Showing More. 
My initial thought was from a general UI we'd want Showing More first. 
Will change.

File: src/components/atoms/buttons/HireMeButton.tsx (hunk 4 above)
Comment (mwilkins74) [reply src/components/atoms/buttons/HireMeButton.tsx:10]: Ok, I'll move this into a Molecule.

Comment (mwilkins74) [reply src/components/atoms/buttons/HireMeButton.tsx:10]: Handling it like this makes more sense and is way more clean.

File: src/components/organisms/videoList/VideoList.tsx [hunk 5]
```diff
@@ -2,18 +2,38 @@ import { Grid } from '@mui/material'
...
   videos: IVideoInfo[]
   pageType: 'genre' | 'role' | 'recent'
 }
 
+const videosPerPage = 3
+
 export function VideoList({ videos, pageType }: VideoListProps) {
-  const videoComponents = videos.map(video => (
+  const location = useLocation()
+  const isHomePage = location.pathname === '/'
```
Comment (ChuckBTaylor) [inline src/components/organisms/videoList/VideoList.tsx:19]: ❓ If the path is: `localhost:8080/#contact`, what does `location.pathname` return? 

Does it return `/` or `/#contact`?

**Review by ChuckBTaylor — APPROVED** [review]

File: src/components/organisms/videoList/VideoList.tsx (hunk 5 above)
Comment (mwilkins74) [reply src/components/organisms/videoList/VideoList.tsx:19]: You were 100% right. 
`Console.log(location.pathname)` is returning `/` when it should be returning `/#contact` here.

**Pull request merged**